### ISSUE TITLE
bug: YT&YTM - Not able to update the app with newer version over old

### ISSUE THREAD

**Issue by vippium** [anddea/revanced-patches#624]
### Type

Cosmetic

### Tools used

rvx-builder

### Application

YT & YTM

### Bug description

I'm not able to update the existing app with the newer version. It always shows "package conflict with existing package". But logically, it shouldn't be like this. If you haven't changed the package name and signature, it shouldn't be.
This error means that signatures are changing at every moment. There should be old behaviour like rvx-manager. Please check at your end too.

### Device Environment

A14 (It doesn't matter)
Issue is regarding builder. 

### Acknowledgements

- [X] This issue does not reproduce on unpatched YouTube or YT Music.
- [X] This issue is not a duplicate of an existing bug report.
- [X] I have chosen an appropriate title.
- [X] All requested information has been provided properly.
- [X] I have written the title and contents in English.

**Comment by AbakNacchan** [comment]
Is the original app installed by system? If it is, try to uninstall the update first. Another possibility might be a problem with signature.

**Comment by vippium** [comment]
> Is the original app installed by system? If it is, try to uninstall the update first. Another possibility might be a problem with signature.

I've removed system yt with adb completely, so it doesn't depend on that. It is something during signing the apk by builder.

**Comment by anddea** [comment]
Seems like you deleted your old keystore, if you use ReVanced and RVX CLIs they have different keystores, so that could be the issue as well.

**Comment by vippium** [comment]
> Seems like you deleted your old keystore, if you use ReVanced and RVX CLIs they have different keystores, so that could be the issue as well.

But it should work after a clean install of both builder and yt. After building from new keystore, I installed it. But after building it again doesn't override the old one.

**Comment by anddea** [comment]
> "package conflict with existing package"

Only occurs if app has the same package name but different keystrore / signature. Do not delete your keystrore or any files generated by builder at this point. You have to ask such questions from community on Reddit or Telegram.

**Comment by vippium** [comment]
> > "package conflict with existing package"
> 
> Only occurs if app has the same package name but different keystrore / signature. Do not delete your keystrore or any files generated by builder at this point. You have to ask such questions from community on Reddit or Telegram.

But I haven't delete any keystore... Let me try again

**Comment by vippium** [comment]
I've also noticed 1 thing in rvx builder. Previously, custom package name are two different patch options. But after inotia's latest update merge, they are merged into one. I checked that changing custom package name doesn't change the package name. It always uses default one (i.e. anddea.xyz). Branding icon working correctly as it still follows old patch options. 
This may be the reason of changing keystore after every build by rvx-builder

**Comment by anddea** [comment]
Custom package name includes both YT and YTM since forever. I make test builds every day, there are no issues with package name or keystore if you do everything right. 

If custom package name is not changing for you, then delete options.json. Probably you're working with wrong files and directories. 

You have to ask such questions from community on Reddit or Telegram.

Use the Manager if you can't figure out what's wrong with the builder. Why do you even use builder in termux instead of revancify?

**Comment by vippium** [comment]
> Custom package name includes both YT and YTM since forever. I make test builds every day, there are no issues with package name or keystore if you do everything right.
> 
> If custom package name is not changing for you, then delete options.json. Probably you're working with wrong files and directories.
> 
> You have to ask such questions from community on Reddit or Telegram.
> 
> Use the Manager if you can't figure out what's wrong with the builder. Why do you even use builder in termux instead of revancify?

Why not... But not till date till manager finds his new ability to download apk's instead of manually downloading them. Just waiting for that day when manager train itself to unlock that ability.
